deepmd/dpmodel/model/linear_model.py

```diff
@@ -0,0 +1,226 @@
+# SPDX-License-Identifier: LGPL-3.0-or-later
+from typing import (
+    Dict,
+    Optional,
+)
+
+import numpy as np
+
+from deepmd.dpmodel import (
+    FittingOutputDef,
+    OutputVariableDef,
+)
+from deepmd.dpmodel.utils.nlist import (
+    build_multiple_neighbor_list,
+)
+
+from .base_atomic_model import (
+    BaseAtomicModel,
+)
+from .dp_atomic_model import (
+    DPAtomicModel,
+)
+from .pair_tab_model import (
+    PairTabModel,
+)
+
+
+class LinearModel(BaseAtomicModel):
+    """Model linearly combine a list of AtomicModels.
+
+    Parameters
+    ----------
+    models
+            This linear model should take a DPAtomicModel and a PairTable model.
+    """
+
+    def __init__(
+        self,
+        dp_models: DPAtomicModel,
+        zbl_model: PairTabModel,
+        **kwargs,
+    ):
+        super().__init__()
+        self.dp_model = dp_models
+        self.zbl_model = zbl_model
+        self.rcut = self.get_rcut()
+        self.sel = self.get_sel()
+
+    def get_rcut(self) -> float:
+        """Get the cut-off radius."""
+        return self.get_rcuts()[-1]
+
+    def get_rcuts(self) -> float:
+        """Get the cut-off radius for each individual models in ascending order."""
+        return sorted([self.zbl_model.get_rcut(), self.dp_model.get_rcut()])
+
+    def get_sel(self) -> int:
+        """Get the neighbor selection."""
+        return self.get_sels()[-1]
+
+    def get_sels(self) -> int:
+        """Get the neighbor selection for each individual models in ascending order."""
+        return sorted([self.zbl_model.get_sel(), sum(self.dp_model.get_sel())])
+
+    def distinguish_types(self) -> bool:
+        """If distinguish different types by sorting."""
+        return self.dp_model.distinguish_types() and self.zbl_model.distinguish_types()
+
+    def forward_atomic(
+        self,
+        extended_coord,
+        extended_atype,
+        nlist,
+        sw_rmin: float,
+        sw_rmax: float,
+        smin_alpha: Optional[float] = 0.1,
```

Comment:
this **breaks** the interface of `forward_atomic`. they are attributes of the class object, please set them at `__init__`

Reply:
fixed.